shared/folders/list.desktop.js

```diff
@@ -1,94 +1,92 @@
 import Row from './row'
 import {globalStyles, globalColors} from '../styles/style-guide'
 
-type State = {
-  showIgnored: boolean
-}
-
 const rowKey = users => users && users.map(u => u.username).join('-')
 
-const Ignored = ({showIgnored, ignored, styles, onToggle, isPublic, onOpen, onClick, onRekey, smallMode}) => {
+const Ignored = ({rows, showIgnored, styles, onToggle, isPublic, onClick}) => {
   const caretIcon: IconType = showIgnored ? 'iconfont-caret-down' : 'iconfont-caret-right'
 
+  if (!rows) {
+    return null
+  }
+
   return (
     <Box style={stylesIgnoreContainer}>
       <Box style={styles.topBox} onClick={onToggle}>
         <Text type='BodySmallSemibold' style={stylesDividerText}>Ignored folders</Text>
-        <Icon type={caretIcon} style={{color: isPublic ? globalColors.black_40 : globalColors.white_40}} />
+        <Icon type={caretIcon} style={{color: isPublic ? globalColors.black_40 : globalColors.white_40, marginTop: -2}} />
       </Box>
       {showIgnored && <Box style={styles.bottomBox}>
-        <Text type='BodySmallSemibold' style={stylesDividerBodyText}>Ignored folders won't show up on your computer and you won't receive alerts about them.</Text>
+        <Text type='BodySmall' style={stylesDividerBodyText}>Ignored folders won't show up on your computer and you won't receive alerts about them.</Text>
       </Box>}
-      {showIgnored && ignored && <Rows
-        rows={ignored}
-        onOpen={onOpen}
-        onClick={onClick}
-        onRekey={onRekey}
-        isPublic={isPublic}
-        smallMode={smallMode}
-        ignored={true}
-      />}
+      {showIgnored && rows}
     </Box>
   )
 }
 
-const Rows = ({rows, ignored, isPublic, onOpen, onClick, onRekey, smallMode}) => (
+const Rows = ({tlfs = [], isIgnored, isPublic, onOpen, onClick, onRekey, smallMode}: Props & {isIgnored: boolean, smallMode?: boolean, tlfs?: Array<Folder>}) => (
   <Box>
-    {!!rows && rows.map((row, idx) => (
-      <Row {...row} key={rowKey(row.users)} isPublic={isPublic} ignored={ignored} onClick={onClick}
-        onRekey={onRekey} onOpen={onOpen} isFirst={!idx} smallMode={smallMode} />
+    {!!tlfs && tlfs.map((tlf) => (
+      <Row
+        {...tlf}
+        key={rowKey(tlf.users)}
+        isPublic={isPublic}
+        ignored={isIgnored}
+        onClick={onClick}
+        onRekey={onRekey}
+        onOpen={onOpen}
+        smallMode={smallMode} />
     ))}
   </Box>
 )
 
-class Render extends Component<void, Props, State> {
-  state: State;
+class Render extends Component<void, Props, void> {
+  _scrollContainer: ?Object;
 
   constructor (props: Props) {
     super(props)
 
-    this.state = {
-      showIgnored: false,
+    this._scrollContainer = null
+  }
+
+  componentDidUpdate (prevProps: Props) {
+    if (prevProps.showIgnored !== this.props.showIgnored &&
+      this._scrollContainer &&
+      this.props.showIgnored) {
+      ReactDOM.findDOMNode(this._scrollContainer).scrollTop += 100
```

Comment:
Design requested an easing animation on this, but the solutions that I can think of to add that would be substantially longer than the implementation here. This version seems to have a fine user experience, as far as I can tell while using it.